kext/FileNVRAM/FileNVRAM.cpp

```diff
@@ -103,6 +103,9 @@ bool FileNVRAM::start(IOService *provider)
     // set a default file path
     setPath(OSString::withCString(FILE_NVRAM_PATH));
     
+    // register immediately, save a lot of startup time
+    registerNVRAM();
```

Comment:
Please do *not* register the nvram module untill after the contents are valid.

For booting w/ a supported boot loader, the means you can do it after the properties are copied.
If the NVRAM is read off of disk, that means it has to be after the callback.

I'd suggest creating a new function, say
FileNVRAM::finalizeInit()
{
    // set mInitComplete = true;
    // registerNVRAM with platform expert
    // other?
}

Any time the code currently sets mInitComplete = true, instead, call the new function.